### PR TITLE
Create some new endpoints and database structures for Comparisons:

### PR DESCRIPTION
**Pull request by joelstobart-moj**
- Add Comparison, Comparison Person, Case Load
- Create a new database model and reflect that in the input data
- Move the security context into a service to be implemented once and extended appropriately. It requires a few test changes

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/ComparisonController.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  )
+  fun create(
+    @RequestBody
+    comparison: ComparisonInput,
+  ): uk.gov.justice.digital.hmpps.calculatereleasedatesapi.entity.Comparison {
+    ComparisonController.log.info("Request received to create a new Comparison")
+    return comparisonService.create(comparison)
+  }
+
+  @GetMapping(value = ["/list/manual"])
```
Comment (jbowdidge-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/ComparisonController.kt:51]: Again, idiomatically, I think this should end up `/comparison/manual`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ServiceUserService.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.apache.commons.lang3.RegExUtils
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.config.AuthAwareAuthenticationToken
+import java.util.stream.Collectors
+
+@Service
+class ServiceUserService {
```
Comment (jbowdidge-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ServiceUserService.kt:12]: This is a really nice encapsulation! I like this a lot!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/ComparisonControllerTest.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    val result = mvc.perform(
+      MockMvcRequestBuilders.get("/compare/$shortReference/count/")
+        .accept(MediaType.APPLICATION_JSON),
+    )
+      .andExpect(MockMvcResultMatchers.status().isOk)
+      .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+      .andReturn()
+
+    Assertions.assertThat(result.response.contentAsString).isEqualTo("7")
```
Comment (jbowdidge-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/ComparisonControllerTest.kt:116]: Can `Assertions` be a static import?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import java.time.LocalDateTime
+import java.util.UUID
+
+@ExtendWith(MockitoExtension::class)
+class ComparisonServiceTest : IntegrationTestBase() {
+
+  private val prisonService = mock<PrisonService>()
+
+  @Autowired
+  lateinit var comparisonService: ComparisonService
```
Comment (jbowdidge-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt:30]: Do you mean `@Autowired` or do you mean `@InjectMocks`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    )
+
+    val comparisonList = alternativeComparisonService.listComparisons()
+
+    Assertions.assertEquals(comparisonList.size, 1)
+  }
+
+  @Test
+  fun `Get a list of manual comparisons`() {
+    SecurityContextHolder.setContext(
```
Comment (jbowdidge-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt:96]: Just intrigued if your new shiny encapsulation could be used as a mock - I haven't followed this bit of code all the way through so I could be wrong.

**Review by jbowdidge-moj — CHANGES_REQUESTED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt (hunk 4 above)
Comment (joelstobart-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt:25]: Done :)

**Review by jbowdidge-moj — APPROVED** [review]
LGTM

**Pull request merged**